packages/faustwp-core/src/components/Toolbar/nodes/Edit.tsx

```diff
@@ -8,17 +9,23 @@ type Props = {
 };
 
 export function Edit({ seedNode }: Props) {
-  if (
-    seedNode === undefined ||
-    seedNode?.isFrontPage ||
-    seedNode?.isPostsPage
-  ) {
+  const {
+    query: { p, typeName },
+  } = useRouter();
+
+  if (seedNode?.isFrontPage || seedNode?.isPostsPage) {
     return null;
   }
 
   // eslint-disable-next-line no-underscore-dangle
-  const postType = seedNode?.__typename || '';
-  const postId = seedNode?.databaseId || '';
+  let postType = seedNode?.__typename || '';
+  let postId = seedNode?.databaseId || '';
+
+  // This is a preview.
+  if (!postId) {
+    postId = p as string;
+    postType = typeName as string;
+  }
```

Comment:
This assumes that if `postId` does not exist, it's a preview, and that `p` and `typeName` do exist. However, that may not always be the case. it's possible to **not** have a `seedNode` and not have a `p`/`typeName` from the URL as well (in the case of CSR rendered pages).

What if we did something like:

```suggestion
  // eslint-disable-next-line no-underscore-dangle
  const postType = seedNode?.__typename ?? (typeName as string | undefined);
  const postId = seedNode?.databaseId ?? (p as string | undefined);

  if (!postId || !postType) {
    return null;
  }
```